### PR TITLE
Pulls in provider records fix from go-ipfs master

### LINKED ISSUE #716: Investigate unresponsive cafes
Possibly related to https://github.com/ipfs/go-ipfs/issues/6237. 

Steps taken so far: 
- increased fd limit 65536 (very high)
- killed websocket swarm handlers (was seeing some writes to closed connections)
- updated the textile service to include:

```
LimitNPROC=32768
LimitNOFILE=65536
Environment="IPFS_FD_MAX=65536"
```

Still, cafes running 0.4.20 have steadily climbing CPU and memory consumption.

### PR DESCRIPTION
**Pull request by sanderpick**
In the name of less configuration, also removes the `Cafe.Host.Local` config flag in favor of the following:

1. Look for a LAN, if present use that IP
2. Poll for a public IP address in the libp2p host's addresses (10s max)
3. Just use loopback

Fixes #716

### CODE REVIEW

**Comment by sanderpick** [comment]
This depends on https://github.com/ipfs/go-ipfs/pull/6289

**Comment by sanderpick** [comment]
Ok, ready for a look on this one even though it's referencing an un-merged go-ipfs PR.

File: mobile/profile.go [hunk 1]
```diff
@@ -13,7 +15,7 @@ func (m *Mobile) Profile() ([]byte, error) {
 
 	self := m.node.Profile()
 	if self == nil {
-		return nil, nil
+		return nil, fmt.Errorf("profile not found")
```
Comment (sanderpick) [inline mobile/profile.go]: @asutula erroring here now.

File: mobile/account.go [hunk 2]
```diff
@@ -45,7 +47,12 @@ func (m *Mobile) AccountContact() ([]byte, error) {
 		return nil, core.ErrStopped
 	}
 
-	return proto.Marshal(m.node.AccountContact())
+	contact := m.node.AccountContact()
+	if contact == nil {
+		return nil, fmt.Errorf("self contact not found")
```
Comment (sanderpick) [inline mobile/account.go]: @asutula erroring here now too.

Comment (asutula) [reply mobile/account.go]: Nice, looks good.

**Comment by sanderpick** [comment]
What do you all think about giving this a shot on the prod cafes? I'd like to see how it behaves even though there are likely other fixes coming/

**Comment by sanderpick** [comment]
The autonat issue was merged. I'll pull that in now.

**Pull request merged**